src/homekit.cs

```diff
@@ -1416,4 +1528,211 @@ interface HMCameraAudioControl
+
+		[Export ("duration")]
+		double Duration { get; set; }
+	}
+
+	[Watch (4,0), TV (11,0), iOS (11,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface HMNumberRange {
+
+		[Static]
+		[Export ("numberRangeWithMinValue:maxValue:")]
+		HMNumberRange FromRange (NSNumber minValue, NSNumber maxValue);
+
+		[Static]
+		[Export ("numberRangeWithMinValue:")]
+		HMNumberRange FromMin (NSNumber minValue);
+
+		[Static]
+		[Export ("numberRangeWithMaxValue:")]
+		HMNumberRange FromMax (NSNumber maxValue);
+
+		[NullAllowed, Export ("minValue", ArgumentSemantic.Strong)]
+		NSNumber Min { get; }
+
+		[NullAllowed, Export ("maxValue", ArgumentSemantic.Strong)]
+		NSNumber Max { get; }
+	}
+
+	[Watch (4,0), TV (11,0), iOS (11,0)]
+	[BaseType (typeof(HMEvent))]
+	[DisableDefaultCtor]
+	interface HMPresenceEvent : NSCopying, NSMutableCopying {
+
+		[Internal]
+		[Export ("initWithPresenceType:")]
+		IntPtr Constructor (NSString presenceType);
+
+		[Wrap ("this (HMPresenceTypeExtensions.GetConstant (presenceType))")]
+		IntPtr Constructor (HMPresenceType presenceType);
+
+		[Internal]
+		[Export ("presenceType")]
+		NSString _PresenceType { get; }
+
+		[Wrap ("HMPresenceTypeExtensions.GetValue (_PresenceType)")]
+		HMPresenceType PresenceType { get; }
+
+		[Field ("HMPresenceKeyPath")]
+		NSString KeyPath { get; }
+	}
+
+	[Watch (4,0), TV (11,0), iOS (11,0)]
+	[BaseType (typeof(HMPresenceEvent))]
+	[DisableDefaultCtor]
+	interface HMMutablePresenceEvent {
+
+		[Internal]
+		[Export ("initWithPresenceType:")]
+		IntPtr Constructor (NSString presenceType);
+
+		[Wrap ("this (HMPresenceTypeExtensions.GetConstant (presenceType))")]
+		IntPtr Constructor (HMPresenceType presenceType);
+
+		[Internal]
+		[Export ("presenceType")]
+		NSString _PresenceType { get; set; }
+
+		// FIXME: Bug https://bugzilla.xamarin.com/show_bug.cgi?id=57870
+		// [Wrap ("HMPresenceTypeExtensions.GetValue (_PresenceType)")]
+		// HMPresenceType PresenceType { get; set; }
+	}
+
+	[Watch (4,0), TV (11,0), iOS (11,0)]
+	[BaseType (typeof(HMTimeEvent))]
+	[DisableDefaultCtor]
+	interface HMSignificantTimeEvent : NSCopying, NSMutableCopying {
+
+		[Internal]
+		[Export ("initWithSignificantEvent:offset:")]
+		IntPtr Constructor (NSString significantEvent, [NullAllowed] NSDateComponents offset);
+
+		[Wrap ("this (HMSignificantEventExtensions.GetConstant (significantEvent), offset)")]
+		IntPtr Constructor (HMSignificantEvent significantEvent, [NullAllowed] NSDateComponents offset);
+
+		[Internal]
+		[Export ("significantEvent", ArgumentSemantic.Strong)]
+		NSString _SignificantEvent { get; }
+
+		[Wrap ("HMSignificantEventExtensions.GetValue (_SignificantEvent)")]
+		HMSignificantEvent SignificantEvent { get; }
+
+		[NullAllowed, Export ("offset", ArgumentSemantic.Strong)]
+		NSDateComponents Offset { get; }
+	}
+
+	[Watch (4,0), TV (11,0), iOS (11,0)]
+	[BaseType (typeof(HMSignificantTimeEvent))]
+	interface HMMutableSignificantTimeEvent {
+
+		[Internal]
```

Comment:
not sure it should be internal, it might be needed in some cases (some NSString are user extensible). 

Maybe just `[Advanced]` to hide it (by default) from intelligence ?